scene/2d/physics_body_2d.cpp

```diff
@@ -1189,7 +1189,7 @@ Vector2 KinematicBody2D::move_and_slide(const Vector2 &p_linear_velocity, const
 				//all is a wall
 				move_and_slide_on_wall = true;
 			} else {
-				if (get_collision_normal().dot(p_floor_direction) > Math::cos(Math::deg2rad((float)45))) { //floor
+				if (get_collision_normal().dot(p_floor_direction) >= Math::cos(Math::deg2rad(p_floor_max_angle))) { //floor
```

Comment:
@akien-mga Well, it you pass `0` as `p_floor_max_angle`, you still want to get matches for exactly perpendicular walls. From my quick testing, it felt much more natural that way.